addons/pos_self_order/tests/test_frontend.py

```diff
@@ -5,32 +5,45 @@
 
 
 @odoo.tests.tagged("post_install", "-at_install")
-class TestFrontend(odoo.tests.HttpCase):
+class TestFrontendMobile(odoo.tests.HttpCase):
+    browser_size = "375x667"
+    touch_enabled = True
+
     def setUp(self):
         super().setUp()
         self.pos_config = self.env["pos.config"].create(
             {
-                "name": "Bar",
+                "name": "BarTest",
                 "module_pos_restaurant": True,
-                "iface_splitbill": True,
-                "iface_printbill": True,
-                "iface_orderline_notes": True,
-                "start_category": True,
+                "is_table_management": True,
                 "self_order_view_mode": True,
+                "self_order_table_mode": False,
             }
         )
-        basic_product = {
-            "name": "Test Product",
+        basic_product = lambda i: {
+            "name": f"Product {i} test",
             "type": "product",
             "available_in_pos": True,
-            "list_price": 10,
+            "list_price": i,
             "taxes_id": False,
         }
-        self.env["product.product"].create([basic_product for i in range(1000)])
+        self.env["product.product"].create([basic_product(i) for i in range(1, 1000)])
+
+    def test_self_order_view_mode_tour(self):
+        self.pos_config.self_order_table_mode = False
+        self.start_tour(
+            self.pos_config._get_self_order_route(),
+            "pos_qr_menu_tour",
+            login=None,
+        )
 
     def test_self_order_tour(self):
+        print(self.pos_config)
+        self.pos_config.self_order_table_mode = True
+        self.pos_config.self_order_pay_after = "meal"
+        self.pos_config.open_ui()
         self.start_tour(
```

Comment:
After this tour, there should be draft orders in the database, right? Kindly add some assertions on the characteristics of those orders after this `start_tour` call.